docs/tutorials/join.md

```diff
@@ -0,0 +1,283 @@
+# Join Primer
+
+This is a brief primer on Zed's experimental [join operator](../zq/operators/join.md).
+
+Currently, join is limited in the following ways:
+* the joined inputs both come from the parent so the query must be split before join,
+* only merge join is implemented requiring inputs to be explicitly sorted, and
+* only equ-join is supported.
+
+A more comprehensive join design with easier-to-use syntax is forthcoming.
+
+## Example Data
+
+The first input data source for our usage examples is `fruit.ndjson`, which describes
+the characteristics of some fresh produce.
+
+```mdtest-input fruit.ndjson
+{"name":"apple","color":"red","flavor":"tart"}
+{"name":"banana","color":"yellow","flavor":"sweet"}
+{"name":"avocado","color":"green","flavor":"savory"}
+{"name":"strawberry","color":"red","flavor":"sweet"}
+{"name":"dates","color":"brown","flavor":"sweet","note":"in season"}
+{"name":"figs","color":"brown","flavor":"plain"}
+```
+
+The other input data source is `people.ndjson`, which describes the traits
+and preferences of some potential eaters of fruit.
+
+```mdtest-input people.ndjson
+{"name":"morgan","age":61,"likes":"tart"}
+{"name":"quinn","age":14,"likes":"sweet","note":"many kids enjoy sweets"}
+{"name":"jessie","age":30,"likes":"plain"}
+{"name":"chris","age":47,"likes":"tart"}
+```
+
+## Inner Join
+
+We'll start by outputting only the fruits liked by at least one person.
+The name of the matching person is copied into a field of a different name in
+the joined results.
+
+Because we're performing an inner join (the default), the inclusion of the
+explicit `inner` is not strictly necessary, but may be included to help make
+the Zed self-documenting.
```

Comment:
```suggestion
Because we're performing an inner join (the default), the
explicit `inner` is not strictly necessary, but including it clarifies our intention.
```